src/services/MillionStatsService.ts

```diff
@@ -35,4 +36,36 @@ export class MillionStatsService {
 
     return holders;  
   }
+
+  static async getPriceData(): Promise<PriceDataMM> {
+    const cacheKey = 'price';
+    const hasCachedData = await cache.has(cacheKey);
+
+    if (hasCachedData) {
+      return await cache.get('price') as PriceDataMM;
+    }
+
+    const apiUrl = `https://api.nomics.com/v1/currencies/ticker?key=${process.env.NOMICS_API_TOKEN}&ids=MM4`;
+    const init: RequestInit = {
+      headers: {
+        'content-type': 'application/json;charset=UTF-8',
+      },
+    }
+    
+    const apiResponse = await fetch(apiUrl, init);
+    const apiResponseBody = await apiResponse.json();
+    const priceChangeRaw = apiResponseBody[0]['1d'].price_change_pct;
```

Comment:
apiResponse may not have the property 'json' as a function and then would lead to an exception.

We're missing any validation of the response body. If the response is not an array, or the first element does not contain '1d' or the value of that does not have price_change_pct you will have an exception.

Similarly the other code in this scope isnt performing any sort of assurance checks to make sure that was is assigned to values are actually as expected. If a typeguard was added then one single check that apiResponse is an apiResponse and a check that it is of the type of response we expect, would handle many of the "this isn't what I expect" exception cases.

I don't want to add a comment for each occurrence but this happens in other places of this PR as well, perhaps we could add a few typeguard to eliminate the bugs? Specifically, the cases that would throw exceptions when handling a value which isn't what was expected.

Reply:
Gotcha. TBH, most of the code was hacked together to get things to just work.  Now that we're starting to add more complexity to the bot, I think the code should be taken more seriously.  Would it be bad practice to just throw 'try catch' blocks everywhere? Also, thoughts on using optional chaining?